test/test_dist_autograd.py

```diff
@@ -159,6 +177,306 @@ def test_rpc_complex_args(self):
                 else:
                     self.assertIsNone(next_funcs[i][0])
 
+    @dist_init
+    def test_error_in_context(self):
+        dst_rank = self._next_rank(self.rank)
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand(3, 3, requires_grad=True)
+            t2 = torch.rand(6, 6, requires_grad=True)
+
+
+            with self.assertRaises(RuntimeError):
+                # This should throw an error.
+                dist.rpc('worker{}'.format(dst_rank), torch.matmul,
+                         args=(t1, t2))
+
+
+    @dist_init
+    def test_backward_simple(self):
+        dst_rank = self._next_rank(self.rank)
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand((3, 3), requires_grad=True)
+            t2 = torch.rand((3, 3), requires_grad=True)
+            ret = dist.rpc('worker{}'.format(dst_rank), torch.add,
+                           args=(t1, t2))
+            self.assertEqual(torch.add(t1, t2), ret)
+            dist_autograd.backward([ret.sum()])
+
+            # Verify grads were accumulated appropriately.
+            grads = dist_autograd.get_gradients(context_id)
+            self.assertEqual(2, len(grads))
+            self.assertIn(t1, grads)
+            self.assertIn(t2, grads)
+            self.assertEqual(torch.ones(3, 3), grads[t1])
+            self.assertEqual(torch.ones(3, 3), grads[t2])
+
+    @dist_init
+    def test_backward_multiple_round_trips(self):
+        dst_rank = self.rank
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand((3, 3), requires_grad=True)
+            t2 = torch.rand((3, 3))
+            t3 = torch.rand((3, 3), requires_grad=True)
+            t4 = torch.rand((3, 3))
+            t5 = torch.rand((3, 3), requires_grad=True)
+
+            # Multiple RPCs between different nodes.
```

Comment:
Err, there's only two nodes involved here right? Because `_next_rank` doesn't mutate any internal state? (Unless I misinterpreted something?)

Reply:
You are right, this was my bad. I have fixed this in the latest commits I pushed.